src/CADL.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -67,88 +67,135 @@ import { OperationPaging } from "./type/OperationPaging.js";
 import { OperationLongRunning } from "./type/OperationLongRunning.js";
 import { OperationFinalStateVia } from "./type/OperationFinalStateVia.js";
 import { getOperationLink } from "@azure-tools/cadl-azure-core";
+import fs from "fs";
+import path from "node:path";
+import { Configuration } from "./type/Configuration.js";
 import { dllFilePath } from "@autorest/csharp";
 import { exec } from "child_process";
 
 export interface NetEmitterOptions {
+    "sdk-folder": string;
     outputFile: string;
     logFile: string;
+    namespace?: string;
+    "library-name"?: string;
+    "shared-source-folders"?: string[];
+    "single-top-level-client"?: boolean;
     skipSDKGeneration: boolean;
     newProject: boolean;
     configurationPath: string;
 }
```

Comment:
Shall we unify the format? 

And we need a flag to let user decide whether to generate condiguration.json. A request from Michael. https://github.com/Azure/autorest.csharp/issues/2783

Reply:
No. Configuration.json is an intermediate file which contains the configurations from cadl-project.yml or cadl options which is needed to generate by emitter and pass to autorest.csharp codegen.

Reply:
what's the format do you mean? the property name? all those config are consistent with autorest.md which is in the flow of swaggger as input.

Reply:
What if user wants to change the SharedSourceFolders?

Reply:
For example, we change outputFile to output-file? Or change "library-name" to libraryName?

Reply:
It should consistent with the name in autorest.md as before.